hazelcast-spring/src/test/resources/com/hazelcast/spring/beans-applicationContext-hazelcast.xml

```diff
@@ -20,9 +20,9 @@
        xmlns:hz="http://www.hazelcast.com/schema/spring"
        xmlns:p="http://www.springframework.org/schema/p"
        xsi:schemaLocation="http://www.springframework.org/schema/beans
-		http://www.springframework.org/schema/beans/spring-beans-3.0.xsd
-		http://www.hazelcast.com/schema/spring
-		http://www.hazelcast.com/schema/spring/hazelcast-spring.xsd">
+        http://www.springframework.org/schema/beans/spring-beans.xsd
+        http://www.hazelcast.com/schema/spring
+        http://www.hazelcast.com/schema/spring/hazelcast-spring-3.10.xsd">
```

Comment:
Instead of a version number, we need to create that file in the first place.

Reply:
There is some resolve strategy how this works that's still a mystery to me and David. Even spring does not have a version-less XSD but it resolved a more recent one. For the hazelcast XSD, a really old one gets resolved and you get validation errors in IDEA. That's why David put the version for the hazelcast XSD but removed it for the spring XSD. I'll try and figure out what's happening here.

Reply:
I believe Spring resolved the link & respond with the latest schema. In our case, we don't have any schema at all, at least `http://www.hazelcast.com/schema/spring/hazelcast-spring.xsd`. But the thing is, IDE should not try to load it from web since the xsd is already in the jar file. I'm also looking.

Reply:
Check this out - https://github.com/hazelcast/hazelcast/issues/4703. The `spring.schemas` file some role here. Also see - https://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/beans/factory/xml/PluggableSchemaResolver.html

Reply:
I think IDEA is the only one here that does not know how to resolve the correct XSD. In the case of the spring XSD, IDEA has probably implemented the resolver like Spring does it. But in the case of hazelcast-spring XSD, it does not have that resolver. Alternatively, you can configure IDEA for the versionless XSD to resolve to the latest XSD in https://www.jetbrains.com/help/idea/schemas-and-dtds.html.
I would therefore remove the version from our XMLs.

Reply:
👍

Reply:
Yes, there is a resolving from URI to files in our in `spring.schemas`:
```
http\://www.hazelcast.com/schema/spring/hazelcast-spring-3.10.xsd=hazelcast-spring-3.10.xsd
http\://www.hazelcast.com/schema/spring/hazelcast-spring.xsd=hazelcast-spring-3.10.xsd
```
But this doesn't seem to work out-of-the-box.

I'll try to add the mapping in IDEA now, let's see if that works out.

Reply:
It should not matter if the file is available online or not, e.g. `http://www.hazelcast.com/schema/sample/hazelcast-sample-service.xsd` works fine as well (after adding the `http://www.hazelcast.com/schema/sample/` in front of it).

It's just the resolution from `hazelcast-spring.xsd` to `hazelcast-spring-3.10.xsd` which doesn't seem to be picked up automatically.

Reply:
Seems the spring `PluggableSchemaResolver` is used to resolve our own schemas as well (respecting our `spring.schemas` file) when the `hazelcast-spring.xml` is being parsed. The IDEA XML editor unfortunately does not do this well. Even the spring schemas are not resolved to the latest schema (they resolve to spring 3.2 schemas instead of 4.3) but we are not getting any errors there since we are not using any more recent spring xml tags.

Reply:
Okay, so this is an IDEA bug? Anyone fancy to test this with Eclipse or NetBeans? :D